### DISCUSSION TITLE
Why is there a limitation on attachment file types?

### DISCUSSION THREAD

**Discussion by typecasto** [community/community#34909] [Projects and Issues]
I'm confused as to why there's a limit on what types of files I'm allowed to attach to an issue or a discussion, and especially why code files aren't allowed. If it was to prevent things like executables from being uploaded, why allow container formats like .gz or .zip? Why is something like an .mp4 acceptable, but not a .py? Why can I upload a .txt, but not a .patch? 

It seems like a needless limitation. I can't attach my window manager's config file to an issue I'm raising, because the config file is written in python. (not like .y(a)ml, .json, .conf, or .ini are allowed either). It's not even preventing me from uploading an executable file, I just have to give it the wrong extension:

[this_is_an_x86_64_gnu_linux_executable.docx](https://github.com/community/community/files/9694030/this_is_an_x86_64_gnu_linux_executable.docx)

(for reference, this is `sp`, a terminal pager, from the `streampager` crate, but it totally could have been malicious)

**Comment by Flamefire** [comment]
> Why can I upload a .txt, but not a .patch?

I'd like to second that uploading text-files such as .patch, .diff, .log should be possible as that is quite often required.

**Comment by mspinder-omegatmm** [comment]
Third this, we currently use Azure to log our bugs but would like to move to github issues. But we often have users submit exports from our program with a .optk extension, these are just text files but with a different extension to play nice with our software.
